components/nexus-repository/src/main/java/org/sonatype/nexus/repository/config/ConfigurationFacetImpl.java

```diff
@@ -89,29 +93,31 @@ public void validate(final Object value, final Class<?>... groups) {
 
     Validator validator = validatorFactory.getValidator();
     Set<ConstraintViolation<Object>> violations = validator.validate(value, groups);
+    ConstraintViolations.maybePropagate(violations, log);
+  }
+
+  /**
+   * Wrap attribute section value to allow configuration hierarchy to be encoded into property path.
+   */
+  private static class SectionWrapper
+  {
+    @Valid
+    private Map<String,Object> attributes = new HashMap<>(1);
```

Comment:
does the map need to be mutable? if not then `Collections.singletonMap(name, value)` would save space